### PR TITLE
Update to aragonOS 4

### PR DESCRIPTION
**Pull request by sohkai**
Completed:

- [x] Restructuring aragonOS section to contain the new and old reference documentation
- [x] aragonOS 4 migration guide
- [x] Small edits to APM introduction
- [x] Move APM reference documentation to APM section
- [x] Update reference docs for aragonOS 4
- [x] Review #48 
- [x] Finish `AragonApp` section
- [x] Run solidity-docgen and add some links to their API documents (e.g. `Kernel`, `ACL`, etc.)

### CODE REVIEW

**Comment by cjyabraham** [comment]
Looks good.  Once this all gets merged into merge-derek-branch2 and then to master, I can do a pass at it to check the language etc. as I have been doing with the rest of hack.aragon.org.

**Comment by sohkai** [comment]
@cjyabraham This is 99% ready to be looked at, I just need to flesh out one section in the reference doc and will push that soon.

File: docs/os-intro.md [hunk 1]
```diff
@@ -7,19 +7,33 @@ hide_title: true
...
+aragonOS does the same for decentralized organizations or protocols. It abstracts away the management of how apps and external entities access the resources of the organization or protocol.
+
+These resources can be things such as assets, cryptocurrencies, the rights to claim a premium on a loan, or the rights to upgrade a smart contract.
+
+Its architecture is based on the idea of a decentralized organization or protocol being the aggregate of multiple components (called applications) connected by a pillar, called the **Kernel**, which are all governed by a special [Access Control List (ACL)](/docs/acl-intro.html) application that controls how these applications and other entities can interact with each other.
+
+--------
 
 aragonOS provides the following functionality:
 - **Upgreadability**: Smart contracts can be **upgraded to a newer version**. Example: *Fixing a bug or adding a feature*.
```
Comment (cjyabraham) [inline docs/os-intro.md]: I think we have too many styles in this section.  I think at least the italics should be removed.

File: docs/os-motivation.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+---
+
+### Why we built aragonOS
+---
+
+aragonOS was born on our path to developing modular and upgradeable smart contracts to power decentralized organizations.
+
+As the project began maturing and we started abstracting more and more, we ended up encapsulating a lot of strong design decisions into a generic framework that could be used by any protocol or decentralized application wanting upgradeability.
+
+**Upgradeability** is really important when developing high-stakes systems in platforms that are still evolving fast. Done well, it can allow for bug fixes and improvements with very little disruption while not causing problems at the protocol level. As a history lesson, if *The DAO* had had an effective upgradeability mechanism, no hard fork would have been required to fix the problem. As a community, we could have avoided months of unproductive discussions, delays in the roadmap, and billions lost in the market capitalization of the protocol due to the fork's damage to network effects.
```
Comment (cjyabraham) [inline docs/os-motivation.md]: could we provide a link to The Dao so it's clear what's being talked about?

**Review by cjyabraham — CHANGES_REQUESTED** [review]
A few comments.  I have not yet reviewed os-3-ref.md and os-ref.md yet but everything else has been reviewed.  I also made a commit (see below) to tighten the language and fix some small things.

File: docs/os-4-migration.md [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+id: aragonos-4-migration
+title: aragonOS 4 migration guide
+sidebar_label: Migrating to aragonOS 4 from aragonOS 3
+---
+
+aragonOS 4 tries very, very hard to be familiar to users of aragonOS 3. In some cases,
+an Aragon app might not even have to be changed when upgrading.
+
+For a more in-depth technical explaination of the changes in aragonOS 4, please see the
+[annoucement blog post (**TODO: LINK**)]() and [reference documentation](/docs/aragonos-ref.html).
```
[the remark this answers is not in the document]
Comment (sohkai) [reply docs/os-4-migration.md]: Has not been, but planning to update it once we get the link URL.

File: docs/apm-ref.md [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+By default a new repo will set the creator (or `dev` param) as the owner of the
+repo and it is the only address that can create new versions in the repo. However,
+as the permission manager, this account can grant permission to other entities to
+create versions. These entities can be anything from another dev, to a multisig or
+a full blown DAO.
+
+## Repos
+
+After discovering an entity in the DAO by traversing the ACL that is an app (see
+section 2.3 *The apps of a DAO*), we can fetch its `app.appId()` and use ENS to
```
[the remark this answers is not in the document]
Comment (sohkai) [reply docs/apm-ref.md]: Ahh, I moved this over from the os-ref document. I'll fix the links in this doc.

File: docs/apm-ref.md [hunk 5]
```diff
@@ -0,0 +1,156 @@
+---
+id: apm-ref
+title: APM reference documentation
+sidebar_label: Reference documentation
+---
+
+##  APM as an Aragon DAO
```
[the remark this answers is not in the document]
Comment (sohkai) [reply docs/apm-ref.md]: See 9cae88b

File: docs/apm-ref.md [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+instance, `aragonpm.eth`, which has high quality standards and strict restrictions on
+what can get published, that we use for publishing our core components.
+
+Different APM registries in which everyone can publish their packages are expected to
+be created by the community, and we have set up `open.aragonpm.eth` on both the main
+and Rinkeby networks as an open instance available for any one to publish to.
+
+This diagram tries to illustrate the architecture of an APM Registry DAO:
+
+![](/docs/assets/apm-arch.jpeg)
```
[the remark this answers is not in the document]
Comment (sohkai) [reply docs/apm-ref.md]: See 9cae88b

File: docs/apm-ref.md [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+```
+
+#### By semantic version
+Providing the exact semantic version.
+```solidity
+repoVersion = repo.getBySemanticVersion([major, minor, patch])
+```
+
+#### By latest contract address
+Fetching the latest version by contract address allows clients to get the latest
```
[the remark this answers is not in the document]
Comment (sohkai) [reply docs/apm-ref.md]: See 25a6682

File: docs/os-intro.md (hunk 1 above)
Comment (sohkai) [reply docs/os-intro.md]: See 73dc3f3.

File: docs/os-motivation.md (hunk 2 above)
Comment (sohkai) [reply docs/os-motivation.md]: See 9369fbe

File: docs/apm-ref.md [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+
+Ownership of the ENS name that the `APMRegistry` operates has to be transferred to
+the `ENSSubdomainRegistrar` app that is part of the DAO.
+
+The `APMRegistry` needs to have permission on the `ENSSubdomainRegistrar` to create
+new names, which is done every time a repo is created.
+
+### APMRegistry governance
+
+Each instance of an APMRegistry can have different governance. Governance of a
```
[the remark this answers is not in the document]
Comment (sohkai) [reply docs/apm-ref.md]: Fixed with f7b0ba0

File: docs/apm-ref.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (sohkai) [reply docs/apm-ref.md]: Fixed with f7b0ba0

**Comment by sohkai** [comment]
@cjyabraham Ready to be reviewed again :). Finished the os-ref document.

File: docs/apm-ref.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (cjyabraham) [reply docs/apm-ref.md]: Some of the text still isn't readable.  I think all font sizes could go up a few notches.  The same goes for [this](https://github.com/aragon/hack/blob/9cae88bb7590384edcb9e2e2a1a2fa90c56b487f/docs/assets/apm-repo.png) diagram.  Actually, would you like me to do an edit of all the diagrams?  I think they could all be improved.  Or do we have a designer that can take a pass at them?

File: docs/os-ref.md [hunk 9]
```diff
@@ -4,615 +4,546 @@ title: aragonOS reference documentation
...
-	- Greater than or equal (`Op.GTE`): Evaluates to true if `args[param.id] >= param.value`.
-	- Less than or equal (`Op.LTE`): Evaluates to true if `args[param.id] <= param.value`.
-	- Return (`Op.RET`): Evaluates to true if `args[param.id]` is greater than one.
-	Used with `PARAM_VALUE_PARAM_ID`, it makes `args[param.id] = param.value`, so it
-	returns the parameter associated value.
-
-While also representing an operation, when the id is `LOGIC_OP_PARAM_ID`, only the
-ops below are valid. These operations use the parameter's value to point to other
-parameters index in the parameter array. These values are encoded as `uint32`
-numbers, left-shifted 32 bits to the left each (example: for example, an op that
```
Comment (cjyabraham) [inline docs/os-ref.md:327]: instead of "left-shifted 32 bits to the left each" how about "each left-shifted 32 bits"?

File: docs/os-ref.md [hunk 10]
```diff
@@ -4,615 +4,546 @@ title: aragonOS reference documentation
...
+> Apps have the choice of which actions to protect behind the ACL, as some actions may make sense to be completely public. However, any publicly exposed state-changing function should *most likely* be protected.
 
-- **Script body:** Address of the contract to make the call to.
-- **Input:** `calldata` for the `delegatecall` that will be performed.
-- **Output:** raw return data of the call.
-- **Blacklist:** impossible to enforce. If there are any addresses in the blacklist the script will revert as it is not possible to check whether a particular address will be called.
+**Lifecycle of an AragonApp call requiring the ACL**
 
-##### 5.2.1.3 DeployDelegateScript
+![](/docs/assets/os-app-call.gif)
```
Comment (cjyabraham) [inline docs/os-ref.md:441]: could we increase the font sizes on this diagram too?

Comment (cjyabraham) [reply docs/os-ref.md:441]: maybe I could work on all the diagrams as a batch?  Do you have the source files for all of them and what application do you use to edit?

File: docs/os-ref.md [hunk 11]
```diff
@@ -4,615 +4,546 @@ title: aragonOS reference documentation
...
+We have designed our own scripting format, known as EVMScripts, to encode complex actions into a bytes representation that can be stored and later executed by another entity. EVMScripts can be installed on a per-organization basis, through a **EVMScriptRegistry**, and aragonOS comes complete with the ability to install multiple script executors in an organization.
 
-The initial version of an app must be a valid bump from version `0.0.0`.
+**EVMScript executors** are contracts that take a script and an input and return an output after execution.
 
-By having this check performed at the smart contract level, we can load the correct
-version of the frontend just by looking at an instance of an app. This is done by
-checking that the version of a smart contract is linked to a given app by getting
-its `appId` and `appCode` (See section *6.3.2.3 By latest contract address*)
+**EVMScript executors and EVMScriptRegistry**
```
Comment (cjyabraham) [inline docs/os-ref.md]: Is this a heading?  I'm confused by it... Is it needed?

File: docs/os-ref.md [hunk 12]
```diff
@@ -4,615 +4,546 @@ title: aragonOS reference documentation
...
 
-```solidity
-repoVersion = (uint16[3] semanticVersion, address contractAddress, bytes contentURI)
-```
+aragonOS provides the `CallsScript` executor as a simple way to concatenate multiple calls. It cancels the operation if any of the calls fail.
 
-#### 6.3.2.1 By versionId
-Every version can be fetched with its `versionId` (which starts in `1` and is
-increments by `1` each version).
+- **Script body:** (See source code file for spec of the payload)
```
Comment (cjyabraham) [inline docs/os-ref.md]: which source code file is this referring to?  can we link to it?

**Review by cjyabraham — CHANGES_REQUESTED** [review]

**Comment by cjyabraham** [comment]
See changeset above for some language changes.  Only os-3-ref.md remains for me to review.

**Review by luisivan — APPROVED** [review]
Looks great to me, I think it should be fine when the stuff @cjyabraham pointed out is solved

File: docs/os-ref.md (hunk 10 above)
Comment (sohkai) [reply docs/os-ref.md:441]: I'm not sure we still have these somewhere, especially since this one was animated... @izqui made them originally and might.

File: docs/os-ref.md [hunk 13]
```diff
@@ -6,28 +6,28 @@ sidebar_label: Reference documentation
...
 
-Additionally, **upgradeability** capabilities are provided and are used by default. aragonOS implements the [DelegateProxy](https://eips.ethereum.org/EIPS/eip-897) pattern, and specifically a special implementation called [unstructured storage](https://blog.zeppelinos.org/upgradeability-using-unstructured-storage/). This pattern essentially splits an contract into two instances: a **base logic contract**, which is then depended upon by a simple, slim **proxy contract**. The proxy delegates all its logic to the linked base contract, and can modify its pointer to the base contract, effectively upgrading its logic.
+Additionally, **upgradeability** capabilities are provided and are used by default. aragonOS implements the [DelegateProxy](https://eips.ethereum.org/EIPS/eip-897) pattern known as  [unstructured storage](https://blog.zeppelinos.org/upgradeability-using-unstructured-storage/). This pattern essentially splits a contract into two instances: a **base logic contract**, which is then depended upon by a simple, slim **proxy contract**. The proxy delegates all its logic to the linked base contract and can modify its pointer to the base contract in order to upgrade its logic.
```
Comment (sohkai) [inline docs/os-ref.md]: @cjyabraham I wouldn't necessarily group unstructured storage with the `DelegateProxy`, as unstructured storage is technically only one possible implementation of the `DelegateProxy` pattern.

File: docs/os-ref.md [hunk 14]
```diff
@@ -458,21 +457,19 @@ This capability is configurable through the overloadable hook:
...
 
-The default implementation of `allowRecoverability()` is just to return true for all tokens, but your overload could choose to not allow certain tokens or even ETH.
+The default implementation of `allowRecoverability()` is just to return true for all tokens but your overload could choose to not allow certain tokens or even ETH.
 
 **Depositable proxies**
 
 AppProxies start off not being able to receive ETH through the native, gas-limited `.send()` and `.transfer()` methods.
 
-However, this can be explicitly enabled through:
+This can be explicitly enabled through this function when an app wants to allow itself (as the proxy instance) to recieve ETH from other contracts using `.send()` or `.transfer()`:
```
Comment (sohkai) [inline docs/os-ref.md]: I would reword this a little bit, due to the duplicate "this"

File: docs/apm-ref.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (sohkai) [reply docs/apm-ref.md]: cc @luisivan could we have Adri take a pass through them?

I don't think we need to wait for this to merge, as it could definitely be done later when Adri has some time.

File: docs/apm-ref.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (sohkai) [reply docs/apm-ref.md]: Mmm yeah, I think we're still not being clear enough then haha.

In more words, the point this bit is trying to say is that each major version is 1-1 related to a contract address. If you query by contract address, you'll get the latest release for that major version (which may not be the latest major version), making it easy to get the latest frontend for an app if you know its contract address.

File: docs/os-ref.md [hunk 15]
```diff
@@ -4,615 +4,546 @@ title: aragonOS reference documentation
...
-function setApp(bytes32 namespace, bytes appId, address app) public;
+function newAppInstance(bytes32 appId, address appBase, bytes initializePayload, bool setDefault);
+function newPinnedAppInstance(bytes32 appId, address appBase, bytes initializePayload, bool setDefault);
 ```
 
-- **Namespace:** specifies what type of app record is being set.
-- **AppId:** used to identify what app is being set. It is the [ENS `namehash`](http://docs.ens.domains/en/latest/introduction.html#namehash) of the APM repo (e.g. `namehash('voting.aragonpm.eth')`).
-- **App:** Address of a contract that can have different meaning depending on the namespace.
+- **initializePayload**: calldata to be used to immediately initialize the app proxy. Useful for atomically initializing the new app proxy in one transaction.
+- **setDefault**: set the new app as the default instance of the app in the Kernel (e.g. also set it in the **App** namespace).
```
[the remark this answers is not in the document]
Comment (sohkai) [reply docs/os-ref.md]: Fixed in 63c5542

File: docs/os-ref.md [hunk 16]
```diff
@@ -4,615 +4,546 @@ title: aragonOS reference documentation
...
 
 ## 4. ACL
 
-A **Permission** is defined as the ability to perform actions (grouped by roles) in a certain app instance (identified by its address).
+A **Permission** is defined as the ability to perform actions (grouped by **Roles**) in a certain app instance (identified by its address).
+
+We refer to a **permission instance** as an entity holding a certain permission. If it helps, you can think of a permission as an _manifestation_ of an app's role that is held by an entity.
 
-We refer to a **Permission Instance** as an entity holding a certain permission.
+### The ACL as an Aragon app
```
[the remark this answers is not in the document]
Comment (sohkai) [reply docs/os-ref.md]: Fixed in 63c5542

File: docs/os-ref.md [hunk 17]
```diff
@@ -4,615 +4,546 @@ title: aragonOS reference documentation
...
-	- `BLOCK_NUMBER_PARAM_ID` (`id = 200`): Sets comparison value to the block number
-	at the time of execution. This allows for setting up timelocks depending
-	on blocks.
-	- `TIMESTAMP_PARAM_ID` (`id = 201`): Sets comparison value to the timestamp of the
-	current block at the time of execution. This allows for setting up timelocks
-	on time.
-	- `SENDER_PARAM_ID` (`id = 202`): Sets comparison value to the sender of the call.
-	(Currently useless because of [this issue]())
-	- `ORACLE_PARAM_ID` (`id = 203`): Checks with an oracle at the address in the
-	*argument value* and returns whether it returned true or false (no comparison with arg).
```
[the remark this answers is not in the document]
Comment (sohkai) [reply docs/os-ref.md:300]: Fixed in 63c5542

File: docs/os-ref.md (hunk 9 above)
Comment (sohkai) [reply docs/os-ref.md:327]: Fixed in 63c5542

File: docs/os-ref.md (hunk 11 above)
Comment (sohkai) [reply docs/os-ref.md]: Fixed in 63c5542 (wasn't needed, good catch!)

File: docs/os-ref.md (hunk 12 above)
Comment (sohkai) [reply docs/os-ref.md]: Fixed in 63c5542

File: docs/apm-ref.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (cjyabraham) [reply docs/apm-ref.md]: I've created a [new issue](https://github.com/aragon/hack/issues/57) for this.

File: docs/apm-ref.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (cjyabraham) [reply docs/apm-ref.md]: I think my suggestion above was unclear.  What I'm suggestion is this (my addition is bolded below):

Fetching the latest version by contract address is useful for clients to easily fetch the latest minor or patch release associated with a particular major version. This makes it particularly easy for organizations to still be served the relevant frontends if they have not updated their smart contract code to the latest **major** version.

File: docs/os-ref.md (hunk 14 above)
Comment (cjyabraham) [reply docs/os-ref.md]: Fixed in 4ad2ab5

File: docs/os-ref.md (hunk 13 above)
Comment (cjyabraham) [reply docs/os-ref.md]: Ok, I've changed to this:
"aragonOS implements the DelegateProxy pattern with a special implementation called unstructured storage."

fixed in 4ad2ab5

**Review by cjyabraham — CHANGES_REQUESTED** [review]

**Comment by cjyabraham** [comment]
I've edited and submitted questions on os-3-ref.  f0b41ab and above.  I've also created [an issue](https://github.com/aragon/hack/issues/57) for improving the diagrams.

File: docs/apm-ref.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (sohkai) [reply docs/apm-ref.md]: Ahh, sorry, misread "final three words" 😅

[the remark this answers is not in the document]
Comment (sohkai) [reply docs/apm-ref.md]: Fixed in 12dc639

File: docs/os-3-ref.md [hunk 18]
```diff
@@ -0,0 +1,423 @@
...
+
+### 1.3 Design philosophy
+
+The design philosophy we use when developing Aragon apps is very similar to the UNIX philosophy, we try to architect them to do one thing and one thing well, and to respect and implement the few aragonOS interfaces so that they play nicely with the rest of the ecosystem.
+
+This results in purely technical benefits such as testability, but it is also very powerful when apps are combined and the output of one app becomes the input of an other one (forwarders resemble UNIX pipes in some way).
+
+### 1.4 Lifecycle of an aragonOS call
+
+![](/docs/assets/os-call.gif)
```
[the remark this answers is not in the document]
Comment (sohkai) [reply docs/os-3-ref.md]: Fixed in ba3bee6

File: docs/os-3-ref.md [hunk 19]
```diff
@@ -0,0 +1,423 @@
...
+decoupling the instance of a particular smart contract with the location of its actual
+business logic. We call individual instances of contracts **Proxy** and the logic
+**base contracts**. A Proxy delegates all its logic on a base contract. Upgradeability
+is achieved because this link to the base contract can be modified, effectively
+updating the Proxy business logic. We created [ERC897](https://github.com/ethereum/EIPs/pull/897) to standardize Proxy interfaces
+for better interoperability in the ecosystem.
+
+- **Forwarder:** A Forwarder is a contract that, given some conditions, will pass
+along a certain action to other contract(s).
+Thanks to the fact that proxies allow a certain instance of a contract to never
```
[the remark this answers is not in the document]
Comment (sohkai) [reply docs/os-3-ref.md:36]: Ahh, yeah this is meant to explain more about how the proxy instance allows you to have a constant identifier (the proxy contract's address) for an application. I've reworded this in ba3bee6.

**Review by izqui — APPROVED** [review]
😍 Amazing push!

File: docs/apm.md [hunk 20]
```diff
@@ -9,37 +9,45 @@ hide_title: true
...
 
-## Architecture: Registries, repos and versions
+A package or **repository** (repo) in an APM registry keeps track of evolving versions of its contents (the webapp component typically) and smart contract code (if applicable).
 
-Each APM Registry is a smart contract that owns an [Ethereum Name Service](https://ens.domains/) (ENS) domain and therefore can create subdomains for its repos. This gives users the ability to **find repos with human-readable names**.
+## Architecture: registries, repos and versions
 
-We envision that many APM registries will be created for different purposes and types of packages. For Aragon, we use the `aragonpm.eth` registry to host our core components. We also have the `open.aragonpm.eth` registry for community developed packages, where anyone can create repos.
+![](/docs/assets/apm-arch.png)
+> The architecture of an APM DAO
```
[the remark this answers is not in the document]
Comment (sohkai) [reply docs/apm.md:27]: Hmm, yeah, on some level it might. cc @cjyabraham should we not use this styling for image notes?

File: docs/stack.md [hunk 21]
```diff
@@ -53,7 +51,7 @@ Which simplified would be:
```
Comment (kanaagawa) [inline docs/stack.md:51]: **"Entity E can call Function F (Param P...) on App A only if Param P satisfies Rules R and Permission Manager PM can revoke or reassign that permission"**

In this statement is not clear:

What those Params are? What they represent?
Where are those Rules?
What is the Permission Manager? (it's also a little confusing name it PM)

Comment (sohkai) [reply docs/stack.md:51]: I've tried to make this a little more clear with 78dae21, although the "rules" are really hard to explain in that context (they get assigned when you create the permission).

File: docs/apm.md (hunk 20 above)
[the remark this answers is not in the document]
Comment (cjyabraham) [reply docs/apm.md:27]: We currently don't have a way to style image captions and notes differently than warning boxes.  For notes and warnings, we use a title to introduce them so it's clear.  Captions usually don't have a title.  I think this might be ok for now but I have [made a note](https://docs.google.com/document/d/1B1nyb95LgBtrWY8Av0FnvnpUfvOb13wfjGZb28svSVs/edit#) that we need to stylistically treat them differently.  We can look at this next month when I work with Delfi on the redesign.

**Review by cjyabraham — APPROVED** [review]

File: docs/os-ref.md [hunk 22]
```diff
@@ -4,615 +4,552 @@ title: aragonOS reference documentation
...
 
-Grants `role` in `app` for an `entity`. Only callable by the `manager` of a certain permission. This `entity` would then be allowed to call all actions that their `role` can perform on that particular `app` until the permission manager revokes their role with [`revokePermission()`](#revoke-permission).
+> **Warning**
+>
+> Only callable by the `manager` of the role's permission instances.
+
+Grants `role` in `app` for an `entity`. This `entity` would then be allowed to call all actions that their `role` can perform on that particular `app` until the permission manager revokes their role with `revokePermission()`.
 
-The `grantPermission()` action doesn’t require protection with the ACL because an entity can only make changes to a permission if it is the permission's `manager`.
+This action is identical to `grantPermission()` except it can only be used by the permission manager of the role.
```
Comment (kanaagawa) [inline docs/os-ref.md]: I think this isn't necessary here

File: docs/os-ref.md [hunk 23]
```diff
@@ -4,615 +4,552 @@ title: aragonOS reference documentation
...
-When assigned to a permission, this rule will **evaluate to true** (and therefore allow the action) if an oracle accepts it and the block number is greater than the previous block number, and either the oracle allows it (again! testing redundancy too) or the first parameter of the rule is lower than 10. The possibilities for customizing organizations/DApps governance model are truly endless, without the need to write any actual Solidity.
+The possibilities for customizing an organization or protocol's governance model are truly endless and there is no need to write any actual Solidity.
+
+### Permissions
+
+`CREATE_PERMISSION_ROLE` protects `createPermission()`.
+
+> **Warning**
+>
+> `CREATE_PERMISSION_ROLE` could be used in malicious and dangerous ways. This is initially assigned when when the Kernel is first initialized. **Protect this permission.**
```
Comment (kanaagawa) [inline docs/os-ref.md]:
```suggestion
> `CREATE_PERMISSION_ROLE` could be used in malicious and dangerous ways. This is initially assigned when the Kernel is first initialized. **Protect this permission.**
```

**Review by 0xGabi — CHANGES_REQUESTED** [review]

File: docs/os-ref.md [hunk 24]
```diff
@@ -4,615 +4,552 @@ title: aragonOS reference documentation
...
 
-By default a new repo will set the creator (or `dev` param) as the owner of the
-repo and it is the only address that can create new versions in the repo. However,
-as the permission manager, this account can grant permission to other entities to
-create versions. These entities can be anything from another dev, to a multisig or
-a full blown DAO.
+**Forwarding and EVMScripts**
 
-### 6.3 Repos
+See the following [Forwarders and EVMScripts](#forwarders-and-evmscripts) section.
```
Comment (kanaagawa) [inline docs/os-ref.md]: I don't think this link is relevant here

File: docs/os-ref.md (hunk 22 above)
Comment (sohkai) [reply docs/os-ref.md]: 😅 oops, forgot to change it to read `createPermission`. Fixed in 7b743b6

File: docs/os-ref.md (hunk 24 above)
Comment (sohkai) [reply docs/os-ref.md]: I think having the link is still useful, since in this context it's not really clear what EVMScripts are for. I've moved the link down to the bottom of the section and reworded it in 7b743b6.
